shell/server/core.js

```diff
@@ -140,17 +141,17 @@ const makeSandstormCore = (grainId) => {
   return new Capnp.Capability(new SandstormCoreImpl(grainId), SandstormCore);
 };
 
-class NotificationHandle {
-  constructor(notificationId, saved, persistentMethods) {
+class NotificationHandle extends PersistentImpl {
+  // TODO(cleanup): Move to a different file.
+
+  constructor(db, saveTemplate, notificationId) {
+    super(db, saveTemplate);
     this.notificationId = notificationId;
-    this.saved = saved;
-    _.extend(this, persistentMethods);
   }
 
   close() {
```

Comment:
Also, I would guess that the reason you're seeing the handle dismiss itself if you don't wrap it is because save() doesn't work because you haven't specified that the object implements the Persistent interface.